packages/flutter/lib/src/cupertino/checkbox.dart

```diff
@@ -0,0 +1,380 @@
+      child: buildToggleable(
+        focusNode: widget.focusNode,
+        autofocus: widget.autofocus,
+        onFocusChange: onFocusChange,
+        size: const Size(48.0, 48.0),
+        painter: _painter
+          ..focusColor = effectiveFocusOverlayColor
+          ..isFocused = focused
+          ..downPosition = downPosition
+          ..activeColor = effectiveActiveColor
+          ..inactiveColor = effectiveInactiveColor
+          ..checkColor = effectiveCheckColor
+          ..value = value
+          ..previousValue = _previousValue
+          ..isActive = widget.onChanged != null
+          ..shape = widget.shape ?? RoundedRectangleBorder(
+            borderRadius: BorderRadius.circular(4.0),
+          )
+          ..side = widget.side,
+      ),
+    );
+  }
+}
+
+const double _kEdgeSize = CupertinoCheckbox.width;
+
+class _CheckboxPainter extends ToggleablePainter {
+  Color get checkColor => _checkColor!;
+  Color? _checkColor;
+  set checkColor(Color value) {
+    if (_checkColor == value) {
+      return;
+    }
+    _checkColor = value;
+    notifyListeners();
+  }
+
+  bool? get value => _value;
+  bool? _value;
+  set value(bool? value) {
+    if (_value == value) {
+      return;
+    }
+    _value = value;
+    notifyListeners();
+  }
+
+  bool? get previousValue => _previousValue;
+  bool? _previousValue;
+  set previousValue(bool? value) {
+    if (_previousValue == value) {
+      return;
+    }
+    _previousValue = value;
+    notifyListeners();
+  }
+
+  OutlinedBorder get shape => _shape!;
+  OutlinedBorder? _shape;
+  set shape(OutlinedBorder value) {
+    if (_shape == value) {
+      return;
+    }
+    _shape = value;
+    notifyListeners();
+  }
+
+  BorderSide? get side => _side;
+  BorderSide? _side;
+  set side(BorderSide? value) {
+    if (_side == value) {
+      return;
+    }
+    _side = value;
+    notifyListeners();
+  }
+
+  Rect _outerRectAt(Offset origin) {
+    const double size = _kEdgeSize;
+    final Rect rect = Rect.fromLTWH(origin.dx, origin.dy, size, size);
+    return rect;
+  }
+
+  // The checkbox's border color if value == false, or its fill color when
+  // value == true or null.
+  Color _colorAt(bool value) {
+    return value && isActive ? activeColor : inactiveColor;
+  }
+
+  // White stroke used to paint the check and dash.
+  Paint _createStrokePaint() {
+    return Paint()
+      ..color = checkColor
+      ..style = PaintingStyle.stroke
+      ..strokeWidth = 2.5
+      ..strokeCap = StrokeCap.round;
+  }
+
+  void _drawBox(Canvas canvas, Rect outer, Paint paint, BorderSide? side, bool fill) {
+    if (fill) {
+      canvas.drawPath(shape.getOuterPath(outer), paint);
+    }
+    if (side != null) {
+      shape.copyWith(side: side).paint(canvas, outer);
+    }
+  }
+
+  void _drawCheck(Canvas canvas, Offset origin, Paint paint) {
+    final Path path = Path();
+    const Offset start = Offset(_kEdgeSize * 0.25, _kEdgeSize * 0.52);
```

Comment:
These numbers in the offsets should be converted to constants to make it easier to read/understand them. Or you could comment on how the values are derived from right here in a comment if this is the only instance of the number.